src/Simplex/Messaging/Agent.hs

```diff
@@ -1912,6 +1935,18 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
                               _ -> pure Nothing
                       _ -> prohibited >> ack
                   _ -> prohibited >> ack
+              updateConnVersion :: Connection c -> ConnData -> Version -> m (Connection c, ConnData)
+              updateConnVersion conn' cData' agentVersion = do
+                aVRange <- asks $ smpAgentVRange . config
+                case versionToRange agentVersion `compatibleVersion` aVRange of
+                  Just _
```

Comment:
you actually need to use the result of compatibleVersion to update the version, you can't just use `agentVersion` as it can be higher than compatible.

```suggestion
                  Just (Compatible v)
```